src/main/java/pl/edu/agh/firecell/engine/algorithm/TemperaturePropagator.java

```diff
@@ -6,187 +6,83 @@
-                oldCell.flammable()) {
-            //set wood on fire
-            newTemperature = WOOD.getBurningTemperature();
-        } else if (burningTime > 0 && oldCell.burningTime() <= MAX_BURNING_TIME) {
-            //keep wood on fire
-            newTemperature = Math.max(WOOD.getBurningTemperature(), currentTemperature);
-        }
-        return newTemperature;
-    }
-
-    private double computeConduction(State oldState, Cell oldCell, Vector3i cellIndex) {
-        return calculateAxisDifference(oldState, NeighbourUtils.north(cellIndex), oldCell, NeighbourUtils.south(cellIndex)) +
+    public double computeConduction(State oldState, Vector3i cellIndex, double currentTemperature) {
+        Cell oldCell = oldState.getCell(cellIndex);
+        return currentTemperature + deltaTime * calculateAxisDifference(oldState, NeighbourUtils.north(cellIndex), oldCell, NeighbourUtils.south(cellIndex)) +
                 calculateAxisDifference(oldState, NeighbourUtils.up(cellIndex), oldCell, NeighbourUtils.down(cellIndex)) +
                 calculateAxisDifference(oldState, NeighbourUtils.east(cellIndex), oldCell, NeighbourUtils.west(cellIndex));
     }
 
     private double calculateAxisDifference(State oldState, Vector3i formerIndex, Cell middleCell, Vector3i furtherIndex) {
-
-        double result = 0.0;
-
-        if (!oldState.hasCell(formerIndex) && !oldState.hasCell(furtherIndex))
-            return result;
-
-        if(oldState.hasCell(formerIndex) && oldState.hasCell(furtherIndex)){
-            Cell formerCell = oldState.getCell(formerIndex);
-            Cell furtherCell = oldState.getCell(furtherIndex);
-            if (formerCell.isSolid() && formerCell.isSolid())
-                result = computeConductivity(formerCell, middleCell, furtherCell);
-            else if (formerCell.isSolid())
-                result = computeConductivity(formerCell, middleCell, new Cell(middleCell));
-            else if (furtherCell.isSolid())
-                result = computeConductivity(new Cell(middleCell), middleCell, furtherCell);
-        } else if(oldState.hasCell(formerIndex) && oldState.getCell(formerIndex).isSolid())
-            result = computeConductivity(oldState.getCell(formerIndex), middleCell, new Cell(middleCell));
-        else if(oldState.hasCell(furtherIndex) && oldState.getCell(furtherIndex).isSolid())
-            result = computeConductivity(new Cell(middleCell), middleCell, oldState.getCell(furtherIndex));
-
-        return result;
+        return oldState.hasCell(formerIndex) && oldState.hasCell(furtherIndex) ?
+                computeConductivity(oldState.getCell(formerIndex), middleCell,
+                        oldState.getCell(furtherIndex),
+                        getCoe(oldState.getCell(formerIndex), middleCell),
+                        getCoe(oldState.getCell(furtherIndex), middleCell)) :
+                0.0;
+        // repair, bug at the border cause one of neighbour is not present
```

Comment:
When cell has precisely one neighbour present, it returns 0, but it should compute the value.